mirgecom/eos.py

```diff
@@ -167,18 +174,33 @@ def get_internal_energy(self, temperature, species_mass_fractions=None):
 
     def dependent_vars(
             self, cv: ConservedVars,
-            temperature_seed: Optional[DOFArray] = None) -> GasDependentVars:
+            temperature_seed: Optional[DOFArray] = None,
+            smoothness_mu: Optional[DOFArray] = None,
+            smoothness_kappa: Optional[DOFArray] = None,
+            smoothness_beta: Optional[DOFArray] = None) -> GasDependentVars:
         """Get an agglomerated array of the dependent variables.
 
         Certain implementations of :class:`GasEOS` (e.g. :class:`MixtureEOS`)
         may raise :exc:`TemperatureSeedMissingError` if *temperature_seed* is not
         given.
         """
         temperature = self.temperature(cv, temperature_seed)
+        # MJA, it doesn't appear that we can have a None field embedded inside DV,
+        # make a dummy smoothness in this case
+        if smoothness_mu is None:
+            smoothness_mu = 0. * cv.mass
```

Comment:
Are there any (compile) risks with initializing `smoothness_mu` from `cv.mass`?  Should we use `zeros` or `zeros_like` directly from the array context?

Reply:
This pattern is used pervasively.  While we prefer the `zeros_like` facility (which was created exactly for this type of thing), this pattern is not (or has not presented itself as) a compilation issue.

Reply:
What do we think of this? [1425245](https://github.com/illinois-ceesd/mirgecom/pull/735/commits/1425245a48ae4b4de54e19fe365944b4eb8d39f2)

Reply:
Looks better -- there are several places where this is used with `smoothness_*`

Reply:
I only found a couple places in eos.   I suppose where they are not passed (i.e. for inviscid-only and runs with AV off), that setting them to the same zeros is OK. [e1ad33f](https://github.com/illinois-ceesd/mirgecom/pull/735/commits/e1ad33fe6c95a551bd40cacaf867baf60c2e015b)